router/src/main.c

```diff
@@ -176,6 +176,46 @@ static void daemon_process(const char *config_path, const char *python_pkgdir,
             //
             umask(0);
 
+
+            //
+            // If config path is not represented by its full path, then
+            // save current path before changing to /
+            //
+            char *config_path_full = NULL;
+            if (strncmp("/", config_path, 1)) {
+                char *cur_path = NULL;
+                int path_size = 256;
+                int getcwd_error = 0;
+
+                cur_path = (char *) malloc(path_size * sizeof(char));
+                memset(cur_path, 0, path_size * sizeof(char));
+
```

Comment:
I'd probably replace these w/ calloc, since the general behavior should be the same as per calloc spec.